core/src/main/java/com/aem/demo/core/services/impl/RssFeedServiceImpl.java

```diff
@@ -41,4 +51,87 @@ public List<NewsCard> readFeed() {
         return feed.getEntries().stream().map(NewsCard::new)
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public String saveRssFeedNodes(List<NewsCard> cardList) {
+        String nodeCreated= StringUtils.EMPTY;
+        try {
+            ResourceResolver resolver = ResolverUtil.newResolver(resourceResolverFactory);
+            Session session = resolver.adaptTo(Session.class);
+
+            if (session.nodeExists(NODE_PATH)){
+                nodeCreated = addNewsCardNode(session, cardList);
+            } else {
+                addParentNode(session);
+                nodeCreated = addNewsCardNode(session, cardList);
+            }
+
+        } catch (Exception e) {
+            log.error("\n Error while creating node - {} ",e.getMessage());
+        }
+        return nodeCreated;
+    }
+
+    private int count = 0;
+
+    private String addNewsCardNode(Session session, List<NewsCard> list){
+
+        try {
+            for (NewsCard elm:list) {
+                count++;
+                String nodeName= count + " newscard_node";
+                Node node = session.getNode(NODE_PATH);
+                Node cardNode = node.addNode(nodeName, NEWSCARDNODE_TYPE);
+                cardNode.setProperty("topic", elm.getTopic());
+                cardNode.setProperty("article", elm.getArticle());
+                cardNode.setProperty("link", elm.getLink());
+                cardNode.setProperty("pubDate", String.valueOf(elm.getPubDate()));
+                cardNode.setProperty("image", elm.getImage());
+                saveImage(cardNode, elm.getImage());
+                session.save();
```

Comment:
It's not the best idea to save the session after every element. The procedure is relatively slow. 

But to be honest, it's not much better to put off saving until some undefined future as well. 

If you knew for sure that the cards are not too many -- to say, there expected to be ~50 cards -- it would be totally allright to do `save` after all of the cards -- just once. 

But if you anticipate there could be hundreds of cards, it's better to do`save` several times -- for instance, when `count % 50 == 0` (that is, after each 50th card), and once more in the very end.